script/train-events.lua

```diff
@@ -131,8 +131,10 @@ function TrainLeaves(trainID)
     if stoppedTrain.train.valid and delivery then
       if delivery.from == stop.entity.backer_name then
         -- update delivery counts to train inventory
+        local original_shipment = {}
         for item, count in pairs (delivery.shipment) do
           local itype, iname = match(item, match_string)
+          original_shipment[item] = count

```

Comment:
Lua passes tables by reference, therefore a copy of the original shipment is needed.

Reply:
Nice double use of this loop.